File: ROS2/Code/chapt2/chapt2_ws/src/demo_python_node/demo_python_node/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self,node_name:str,name_value: str,age_value: int) ->None:
        
        super().__init__(node_name)
        print('PersonNode __init__')
        self.name = name_value
        self.age = age_value

    def eat(self,food_name: str):
        """
        方法：吃东西
        :food_name 食物名字
        """
        self.get_logger().info(f"{self.name},{self.age}岁,爱吃{food_name}")
        # print(f"{self.name},{self.age}岁,爱吃{food_name}")


def main():
    rclpy.init()

    node = PersonNode("xiaowang","小王",36)
    node.eat("紫菜蛋花汤")

    rclpy.spin(node)
    rclpy.shutdown()
    

File: ROS2/Code/chapt2/chapt2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"

class PersonNode : public rclcpp::Node
{
private:
    std::string name_;
    int age_;

public:
    PersonNode(const std::string &node_name, const std::string &name, const int &age) : Node(node_name)
    {
        this->name_ = name;
        this->age_ = age;
    };

    void eat(const std::string &food_name)
    {

        RCLCPP_INFO(this->get_logger(), "%s,%d岁,爱吃%s", this->name_.c_str(), this->age_, food_name.c_str());
    };
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("person_node","小王",36);
    node->eat("紫菜蛋花汤");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
